pkg/utils/tsoutil/tso_dispatcher.go

```diff
@@ -208,8 +214,10 @@ func NewTSDeadline(
 	done chan struct{},
 	cancel context.CancelFunc,
 ) *TSDeadline {
+	timer := timerPool.Get().(*time.Timer)
+	timer.Reset(timeout)
 	return &TSDeadline{
```

Comment:
How about we use sync.Pool for TSDeadline and a timer is always attached to a TSDeadline and can be reused and reset every time? In this way, we use sync.Pool for both TSDeadline and Timer in fact.

Reply:
It will make code complex and test show that `done` and `cancel` don't cost some memory.

Reply:
You're right. Timer is the main issue and we can keep smaller change.